libs/sdk-ui-tests/stories/_infra/generateInsightStories.ts

```diff
@@ -0,0 +1,214 @@
+     * recording index directly when building storybook...
+     */
+    return values(recordings.metadata?.insights ?? {}).map((rec) => rec.obj);
+}
+
+const Insights = [...getAvailableInsights(ReferenceRecordings.Recordings)];
+
+const InsightById = keyBy(Insights, insightId);
+
+const ScenarioGroupsByVis = sortBy(
+    values(groupBy<ScenarioGroup<any>>(AllTestScenarioGroups, (g) => g.vis)),
+    (groups) => groups[0].vis,
+);
+
+ScenarioGroupsByVis.forEach((groups, groupsIndex) => {
+    /*
+     * Sort groups; the order in which stories for the group are created is important as that is the order
+     * in which the groups appear in storybook.
+     */
+    const sortedGroups = sortBy(groups, (g) => g.groupNames.join("/"));
+
+    // eslint-disable-next-line sonarjs/cognitive-complexity
+    sortedGroups.forEach((group, groupIndex) => {
+        // because we have duplicate group names, and duplicate scenario names within those duplicate group names
+        // we have to create a unique file if one with the same name exists, otherwise they will overwrite each other
+
+        let groupName = "";
+        let fileDirectory = "";
+        let filePath = "";
+        let groupNameModifier = "-01";
+        let uniqueFilePathFound = false;
+        while (!uniqueFilePathFound) {
+            let groupNameParts = [group.vis, ...group.groupNames];
+
+            groupName = ["04 Stories For Pluggable Vis", ...groupNameParts].join("/");
+
+            groupNameParts[groupNameParts.length - 1] =
+                groupNameParts[groupNameParts.length - 1] + groupNameModifier;
+            groupNameParts = groupNameParts.map((g) => g.replaceAll(/\s+/g, "-"));
+
+            const fileName = `${groupNameParts.pop()}.generated.stories.tsx`;
+            fileDirectory = `./stories/visual-regression/visualizations/insightStories/${
+                groupNameParts.join("/") + (groupNameParts.length > 0 ? "/" : "")
+            }`;
+            filePath = `${fileDirectory}${fileName}`;
+
+            if (!existsSync(filePath)) {
+                uniqueFilePathFound = true;
+                break;
+            }
+
+            groupNameModifier =
+                "-" + (parseInt(groupNameModifier.substring(1)) + 1).toString().padStart(2, "0");
+        }
+
+        const defaultExport = `export default {
+    title: "${groupName}",
+};`;
+
+        const filePathDepth = filePath.split("/").length - 2; // one at the start and one at the end
+        const pathToRoot = "../".repeat(filePathDepth);
+
+        const requires = generateImports([
+            {
+                source: "@gooddata/sdk-model",
+                namedImports: ["IInsight"],
+            },
+            {
+                source: `${pathToRoot}stories/visual-regression/visualizations/insightStories.js`,
+                namedImports: ["plugVizStory", "getScenariosGroupByIndexes"],
+            },
+            // {
```

Comment:
Do we miss some stories here?

`${pathToRoot}/src/scenario.js` ?

If not pls remove this comment